src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplateSupport.java

```diff
@@ -82,4 +103,74 @@ public String getJavaNameForEntity(final Class<?> clazz) {
 		return new ConvertingPropertyAccessor<>(accessor, converter.getConversionService());
 	}
 
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.applicationContext = applicationContext;
+		if (entityCallbacks == null) {
+			setEntityCallbacks(EntityCallbacks.create(applicationContext));
+		}
+	}
+
+	/*
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.eventPublisher = applicationEventPublisher;
+	}
+	 */
+
+	/**
+	 * Set the {@link EntityCallbacks} instance to use when invoking
+	 * {@link org.springframework.data.mapping.callback.EntityCallback callbacks} like the {@link BeforeConvertCallback}.
+	 * <p/>
+	 * Overrides potentially existing {@link EntityCallbacks}.
+	 *
+	 * @param entityCallbacks must not be {@literal null}.
+	 * @throws IllegalArgumentException if the given instance is {@literal null}.
+	 * @since 2.2
+	 */
+	public void setEntityCallbacks(EntityCallbacks entityCallbacks) {
+		Assert.notNull(entityCallbacks, "EntityCallbacks must not be null!");
+		this.entityCallbacks = entityCallbacks;
+	}
+
+	void maybeEmitEvent(CouchbaseMappingEvent<?> event) {
```

Comment:
formatting is way off here

Reply:
I'm using https://github.com/krasa/EclipseCodeFormatter#instructions. Maybe I have the wrong settings for it?